ouroboros-consensus/test/consensus-test/Test/Consensus/HardFork/Summary.hs

```diff
@@ -236,7 +270,7 @@ instance Arbitrary ArbitrarySummary where
         beforeHorizonSeconds <-
           choose (0, summaryTimeSpanSeconds)
             `suchThat` \x -> x /= summaryTimeSpanSeconds
-
+        beforeHorizonPerasRounds <- choose (0, summaryPerasRounds - 1) -- Note: this will underflow if summaryPerasRounds is 0
```

Comment:
AFAICT we never actually generate inputs in such a way that `summaryPerasRounds == 0`, so maybe `error` here if that is the case (maybe due to a future change)?

Also: overly long line